snapcraft/wrappers/gpu-2404-custom-wrapper

```diff
@@ -5,6 +5,7 @@
 if snapctl is-connected gpu-2404
 then
     echo "INFO: the gpu-2404 interface is connected. Running with gpu-2404 wrapper."
+    export XDG_DATA_DIRS=${XDG_DATA_DIRS:+$XDG_DATA_DIRS:}/usr/share
```

Comment:
Please can you add a comment here explaining why this is needed.

What is the "+" doing here btw?

Reply:
>What is the "+" doing here btw?

Hmm, I'm using the same way we have in many other places.
But strictly speaking, it is (from bash man pages):
```
       ${parameter:+word}
              Use Alternate Value.  If parameter is null or unset, nothing is substituted, otherwise
              the expansion of word is substituted.
```

Reply:
>Please can you add a comment here explaining why this is needed.

Have done.

Reply:
Thanks.

I'm struggling to see what this change is doing, or rather, why it helps.

You seem to be just adding XDG_DATA_DIRS to itself and then `/usr/share` too. Am I interpreting this correctly?

Reply:
Was `/usr/share` missing from the XDG_DATA_DIRS value set externally?

Reply:
>You seem to be just adding XDG_DATA_DIRS to itself and then /usr/share too. Am I interpreting this correctly?

Yeah, precisely.

>Was /usr/share missing from the XDG_DATA_DIRS value set externally?

Yep. It's not there and we add it.

All paths in `XDG_DATA_DIRS` are adjusted with a snap prefix like `/snap/lxd/<rev>/gpu-2404-2/<...>/usr/share` and this is correct. But nvidia-container-toolkit tries to append a  `driverRoot` value as a prefix which leads to a situation when these path became incorrect. By adding `/usr/share` I workaround this.

Reply:
Thanks for the info. And thanks for opening the upstream bug fix too.